sql/rpl_reporting.h

```diff
@@ -101,24 +101,7 @@ class Slave_reporting_capability
 
     }
 
-    void update_timestamp()
-    {
-      struct tm tm_tmp;
-      struct tm *start;
-
-      skr= my_time(0);
-      localtime_r(&skr, &tm_tmp);
-      start=&tm_tmp;
-
-      sprintf(timestamp, "%02d%02d%02d %02d:%02d:%02d",
```

Comment:
What was fixed there?

Reply:
With sprintf, gcc belives that this could overflow (it can't, as it's 15 chars max + null terminator). So I changed it to my_snprintf, which silenced the warning, but that function isn't available in the header, so I moved the implementation to the cpp file.

(and this is a point where mysql fixed this differently: they simply increased the buffer size to 64 bytes... which just makes no sense)